src/collaboration/push-protocol.js

```diff
@@ -28,20 +29,40 @@ module.exports = class PushProtocol {
   }
 
   forPeer (peerInfo) {
+    const dbg = (...args) => debug('%s: ' + args[0], this._peerId(), ...args.slice(1))
     const remotePeerId = peerInfo.id.toB58String()
-    debug('%s: push protocol to %s', this._peerId(), remotePeerId)
+    dbg('push protocol to %s', remotePeerId)
 
     const queue = new Queue({ concurrency: 1 })
     let ended = false
+    // Is the connection in eager (pushing) mode
     let pushing = true
-    let isPinner = false
+    // Last clock sent to the remote peer
     let sentClock = {}
+    // Our copy of the remote peer's clock
     let remoteClock = {}
 
-    const sendClockDiff = (clock) => {
+    // Is the remote peer a pinner
+    let isPinner
+    // To find out if the remote peer is a pinner, we need to wait till it
+    // sends us a message
+    let pinnerInfoEmitter = new EventEmitter()
```

Comment:
Maybe a good opportunity to use the `p-event` package? https://www.npmjs.com/package/p-event